src/roiextractors/extractors/tiffimagingextractors/scanimagetiffimagingextractor.py

```diff
@@ -22,13 +24,487 @@ def _get_scanimage_reader() -> type:
+        Path to the TIFF file.
+
+    Returns
+    -------
+    extra_metadata: dict
+        Dictionary of metadata extracted from the TIFF file.
+    """
+    ScanImageTiffReader = _get_scanimage_reader()
+    io = ScanImageTiffReader(str(file_path))
+    extra_metadata = {}
+    for metadata_string in (io.description(iframe=0), io.metadata()):
+        metadata_dict = {
+            x.split("=")[0].strip(): x.split("=")[1].strip()
+            for x in metadata_string.replace("\n", "\r").split("\r")
+            if "=" in x
+        }
+        extra_metadata = dict(**extra_metadata, **metadata_dict)
+    return extra_metadata
+
+
+def parse_metadata(metadata):
+    """Parse metadata dictionary to extract relevant information and store it standard keys for ImagingExtractors.
+
+    Currently supports
+    - sampling_frequency
+    - num_channels
+    - num_planes
+    - frames_per_slice
+    - channel_names
+
+    Parameters
+    ----------
+    metadata : dict
+        Dictionary of metadata extracted from the TIFF file.
+
+    Returns
+    -------
+    metadata_parsed: dict
+        Dictionary of parsed metadata.
+
+    Notes
+    -----
+    SI.hChannels.channelsActive = '[1;2;...;N]' where N is the number of active channels.
+    SI.hChannels.channelName = "{'channel_name_1' 'channel_name_2' ... 'channel_name_M'}"
+        where M is the number of channels (active or not).
+    """
+    sampling_frequency = float(metadata["SI.hRoiManager.scanVolumeRate"])
+    num_channels = len(metadata["SI.hChannels.channelsActive"].split(";"))
+    num_planes = int(metadata["SI.hStackManager.numSlices"])
+    frames_per_slice = int(metadata["SI.hStackManager.framesPerSlice"])
+    channel_names = metadata["SI.hChannels.channelName"].split("'")[1::2][:num_channels]
+    metadata_parsed = dict(
+        sampling_frequency=sampling_frequency,
+        num_channels=num_channels,
+        num_planes=num_planes,
+        frames_per_slice=frames_per_slice,
+        channel_names=channel_names,
+    )
+    return metadata_parsed
+
+
+def parse_metadata_v3_8(metadata):
+    """Parse metadata dictionary to extract relevant information and store it standard keys for ImagingExtractors.
+
+    Requires old version of metadata (v3.8).
+    Currently supports
+    - sampling frequency
+    - num_channels
+    - num_planes
+
+    Parameters
+    ----------
+    metadata : dict
+        Dictionary of metadata extracted from the TIFF file.
+
+    Returns
+    -------
+    metadata_parsed: dict
+        Dictionary of parsed metadata.
+    """
+    sampling_frequency = float(metadata["state.acq.frameRate"])
+    num_channels = int(metadata["state.acq.numberOfChannelsSave"])
+    num_planes = int(metadata["state.acq.numberOfZSlices"])
+    metadata_parsed = dict(
+        sampling_frequency=sampling_frequency,
+        num_channels=num_channels,
+        num_planes=num_planes,
+    )
+    return metadata_parsed
+
+
+class MultiPlaneImagingExtractor(ImagingExtractor):
```

Comment:
I would say move this to the outer level next to the base classes

Reply:
And maybe add a TODO to rename the previous `MultiFrameImagingExtractor`?

Reply:
To help disambiguate

Reply:
> I plan on moving the MultiPlaneImagingExtractor to a different file and a different PR, but I included it to give a sense of how I see the ScanImageExtractors working with the inheritance structure.

Yes.

> And maybe add a TODO to rename the previous MultiFrameImagingExtractor?

Yeah I was thinking about that, but didn't want to break old code that used `MultiImagingExtractor`... maybe we can deprecate it and remove in 3mo's like `ScanImageTiffImagingExtractor`.  Also, MultiFrame isn't quite right bc single-file IE's can have multiple frames but not need `MultiImagingExtractor`.  Maybe MultiFile or MultiChunk?

Reply:
It's not MultiFile either, you might want to concatenate reshaped contents from a single file. Chunk is an overloaded term everywhere anyway so I'd avoid using that anywhere

MultiTimeImagingExtractor? TimeConcatenatedImagingExtractor?

Reply:
Ok, after thinking a bit about this, I think the best approach is to rename the `MultiPlaneImagingExtractor` to `VolumetricImagingExtractor` or something like that to avoid making the comparison with `MultiImagingExtractor` since they're not quite parallel.  Then we can leave `MultiImagingExtractor` as is or rename it `MultiFrameImagingExtractor` to make it extra clear that we are concatenating along the frames axis.

Reply:
I dislike the profileration of specific classess to cut through channels and planes but `VolumetricImagingExtractor` is a great name. 

I think that a useful heuristic is that classes names should reflect to users what they are for and not implementation details. In this case, `MultiImagingExtractor` seems to be kind of doing the latter as we are leaking to the user that we are building this extractor out of multiple ones but whey they should care about this?

If give you my format, you give me an object that allows me to extract my data ergonomically and efficiently with a common API. I don't need to know **how** you do it.

Reply:
`VolumetricImagingExtractor` sounds great 👍

I'm fine with either `MultiImagingExtractor` or `MultiFrameImagingExtractor` in conjunction with that

Reply:
Moving this to a dedicated PR #248